### PR TITLE
[safety-rules] Improve QC handling

### PR DESCRIPTION
**Pull request by davidiw**
* Unify the 3 methods that care about LIs/QCs to just 2
* Verify QCs and BlockInfos are sane -- probably should also add in LIs to that mix -- next PR!

Because we now verify QCs, SafetyRules /must/ be initialized properly so a good chunk of the PR is spent making our tests more functionally accurate.

### CODE REVIEW

File: consensus/safety-rules/src/safety_rules.rs [hunk 1]
```diff
@@ -82,13 +82,31 @@ impl<T: Payload> SafetyRules<T> {
...
+        let validator_verifier = self
+            .validator_verifier
+            .as_ref()
+            .ok_or(Error::NotInitialized)?;
+
+        qc.verify(validator_verifier)
+            .map_err(|e| Error::InvalidQuorumCertificate(e.to_string()))?;
+
+        if qc.parent_block().round() < self.persistent_storage.preferred_round()? {
+            Err(Error::InvalidQuorumCertificate(
```
Comment (zekun000) [inline consensus/safety-rules/src/safety_rules.rs:97]: this shouldn't be an error, just ignore it otherwise caller has to ensure this

Comment (davidiw) [reply consensus/safety-rules/src/safety_rules.rs:97]: Shouldn't the caller?

File: consensus/consensus-types/src/quorum_cert.rs [hunk 2]
```diff
@@ -123,6 +123,7 @@ impl QuorumCert {
         self.ledger_info()
             .verify_signatures(validator)
             .context("Fail to verify QuorumCert")?;
+        self.vote_data.verify()?;
```
[the remark this answers is not in the document]
Comment (davidiw) [reply consensus/consensus-types/src/quorum_cert.rs:126]: Next PR, want to minimize integration test breaks :)

File: consensus/safety-rules/src/safety_rules.rs (hunk 1 above)
Comment (davidiw) [reply consensus/safety-rules/src/safety_rules.rs:97]: This shouldn't have any ramifications as is, so landing without modification. Will adjust the caller to differentiate liveness, safety, and warning "errors"

**Pull request closed without merging**